src/Farmer/Builders/Builders.Dns.fs

```diff
@@ -161,6 +174,88 @@ type DnsMxRecordBuilder() =
     [<CustomOperation "ttl">]
     member _.RecordTTL(state:MxRecordProperties, ttl) = { state with TTL = Some ttl }
 
+type DnsSrvRecordBuilder() =
+    member __.Yield _ = { SrvRecordProperties.Name = ResourceName "@"; SrvValues = []; TTL = None; }
+    member __.Run(state : SrvRecordProperties) = DnsZoneRecordConfig.Create(state.Name, state.TTL, SRV state.SrvValues)
+
+    /// Sets the name of the record set.
+    [<CustomOperation "name">]
+    member _.RecordName(state:SrvRecordProperties, name) = { state with Name = name }
+    member this.RecordName(state:SrvRecordProperties, name:string) = this.RecordName(state, ResourceName name)
+
+    /// Add SRV records.
+    [<CustomOperation "add_values">]
+    member _.RecordValue(state:SrvRecordProperties, srvValues : SrvRecord list) =
+        { state with SrvValues = state.SrvValues @ srvValues }
+
+    /// Sets the TTL of the record.
+    [<CustomOperation "ttl">]
+    member _.RecordTTL(state:SrvRecordProperties, ttl) = { state with TTL = Some ttl }
+
+type DnsSoaRecordBuilder() =
+    member __.Yield _ = 
+        { SoaRecordProperties.Name = ResourceName "@"
+          Host = None
+          Email = None
+          SerialNumber = None
+          RefreshTime = None
+          RetryTime = None
+          ExpireTime = None
+          MinimumTTL = None
+          TTL = None }
+
+    member __.Run(state : SoaRecordProperties) = 
+        let value = 
+            { Host = state.Host
+              Email = state.Email
+              SerialNumber = state.SerialNumber
+              RefreshTime = state.RefreshTime
+              RetryTime = state.RetryTime
+              ExpireTime = state.ExpireTime
+              MinimumTTL = state.MinimumTTL }
+        DnsZoneRecordConfig.Create(state.Name, state.TTL, SOA value)
+
+    /// Sets the name of the record set.
+    [<CustomOperation "name">]
+    member _.RecordName(state:SoaRecordProperties, name) = { state with Name = name }
+    member this.RecordName(state:SoaRecordProperties, name:string) = this.RecordName(state, ResourceName name)
+
+    /// Sets the name of the record set.
+    [<CustomOperation "host">]
+    member _.RecordHost(state:SoaRecordProperties, host : string) = { state with Host = Some host }
+
+    /// Sets the email for this SOA record (required).
+    [<CustomOperation "email">]
+    member _.RecordEmail(state:SoaRecordProperties, email : string) = { state with Email = Some email }
+
+    /// Sets the expire time for this SOA record in seconds. 
+    /// Defaults to 2419200 (28 days).
+    [<CustomOperation "expire_time">]
+    member _.RecordExpireTime(state:SoaRecordProperties, expireTime : int64) = { state with ExpireTime = Some expireTime }
+
+    /// Sets the minimum time to live for this SOA record in seconds.
+    /// Defaults to 300.
+    [<CustomOperation "minimum_TTL">]
+    member _.RecordMinimumTTL(state:SoaRecordProperties, minTTL : int64) = { state with MinimumTTL = Some minTTL }
```

Comment:
Can you create an issue to replace the use of int64 (across *all* the builders in DNS Zone) to use overloads of either `<second>` or `TimeSpan`? We can deal with that as a separate PR.